### PR TITLE
Hibernate-5-ify Document (formerly cms_attachment)

### PR DESCRIPTION
**Pull request by jasonaowen**
Remove redundant comments, empty constructor, and trailing whitespace (I should just make a text snippet for this...); rename the table to reflect the name of the class, pluralize the table name, and use Hibernate to generate IDs.

Issue #36 Use Hibernate 5, instead of 4

### CODE REVIEW

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java [hunk 1]
```diff
@@ -1160,11 +1159,12 @@ private void insertAffiliations(ProviderProfile details, Map<Long, Long> attachm
             if (attachment.getId() > 0) {
                 // regenerate attachment id when cloning, but keep reference so we can
                 // fix non-mapped references
-                long generated = getSequence().getNextValue(Sequences.ATTACHMENT_SEQ);
-                mapping.put(attachment.getId(), generated);
-                attachment.setId(generated);
+                long oldAttachmentId = attachment.getId();
+                long newAttachmentId = uploadAttachment(attachment);
+                mapping.put(oldAttachmentId, newAttachmentId);
```
Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: Is "cloning" when an attachment is replaced?  I'm trying to understand when we would get a new attachment id.

Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: Yeah... and now that I'm able to get further in the process, this code isn't doing the right think, and will need to change.

I don't entirely understand the flow yet, but it seems to be something like: enrollments get duplicated as they go through the phases of draft -> submitted -> approved. I need to look into this more deeply.

Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: Oh, strange, okay.  Clearly figuring that out is beyond the scope of this PR!

Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: I figure introducing new code that doesn't work is not a great practice... Fortunately, it was a small change, once I was able to focus on it! Fixed.

**Pull request merged**